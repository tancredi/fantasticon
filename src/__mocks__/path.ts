const _path = jest.requireActual('path');
const _relative = _path.relative;

const projectDir = _path.resolve(__dirname, '../../');

const resolve = (...paths: string[]) => {
  let path = '';

  for (const cur of paths) {
    path = !path ? normalize(cur) : _path.join(path, normalize(cur));
  }

  if (path.startsWith(projectDir)) {
    path = _path.join('/root/project', path.substr(projectDir.length));
  } else if (path.startsWith('/') && !path.startsWith('/root')) {
    path = _path.join('/root/', path.substr(1));
  } else if (path.startsWith('./')) {
    path = _path.join('/root/project/', path.slice(2));
  } else if (!path.startsWith('/')) {
    path = _path.join('/root/project/', path);
  }

  return normalize(path);
};

const relative = (a: string, b: string) =>
  normalize(_relative(_path.normalize(a), _path.normalize(b)));

const join = (...segments: string[]): string => {
  const trimmed: string[] = [];

  for (let [i, current] of segments.entries()) {
    const isFirst = i === 0;
    const isLast = i === segments.length - 1;

    if (!isFirst && current.startsWith('/')) {
      current = current.substr(1);
    }

    if (!isLast && current.endsWith('/')) {
      current = current.substring(0, current.length - 1);
    }

    trimmed.push(current);
  }

  return trimmed.join('/');
};

const normalize = (path: string) => path.replace(/\\/g, '/');

const isAbsolute = (path: string) => path.startsWith('/root');

module.exports = { resolve, relative, join, isAbsolute, normalize };
