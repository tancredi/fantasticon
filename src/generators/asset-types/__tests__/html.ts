import { resolve } from 'path';
import htmlGen from '../html';

const mockOptions = {
  name: 'test-font',
  prefix: 'tf',
  tag: 'b',
  assets: { 'my-icon': null },
  templates: {
    html: resolve(__dirname, '../../../../templates/html.hbs')
  }
} as any;

describe('`HTML` asset generator', () => {
  test('renders HTML correctly with prefix and tag name options', async () => {
    expect(await htmlGen.generate(mockOptions, null)).toMatchSnapshot();
  });

  test('rendered HTML contains expected CSS path', async () => {
    expect(await htmlGen.generate(mockOptions, null)).toContain(
      '"test-font.css"'
    );
  });

  test('rendered HTML contains expected tag and class name', async () => {
    expect(await htmlGen.generate(mockOptions, null)).toContain(
      '<b class="tf tf-my-icon"></b>'
    );
  });

  test('rendered HTML contains expected title', async () => {
    expect(await htmlGen.generate(mockOptions, null)).toContain(
      `<h1>test-font</h1>`
    );
  });
});
