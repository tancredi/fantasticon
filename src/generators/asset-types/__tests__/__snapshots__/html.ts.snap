// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`HTML\` asset generator renders HTML correctly with prefix and tag name options 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
    <meta charset=\\"UTF-8\\">
    <title>test-font</title>

    <style>
        body {
            font-family: sans-serif;
            margin: 0;
            padding: 10px 20px;
            text-align: center;
        }
        .preview {
            width: 100px;
            display: inline-block;
            margin: 10px;
        }
        .preview .inner {
            display: inline-block;
            width: 100%;
            text-align: center;
            background: #f5f5f5;
            -webkit-border-radius: 3px 3px 0 0;
            -moz-border-radius: 3px 3px 0 0;
            border-radius: 3px 3px 0 0;
        }
        .preview .inner  {
            line-height: 85px;
            font-size: 40px;
            color: #333;
        }
        .label {
            display: inline-block;
            width: 100%;
            box-sizing: border-box;
            padding: 5px;
            font-size: 10px;
            font-family: Monaco, monospace;
            color: #666;
            white-space: nowrap;
            overflow: hidden;
            text-overflow: ellipsis;
            background: #ddd;
            -webkit-border-radius: 0 0 3px 3px;
            -moz-border-radius: 0 0 3px 3px;
            border-radius: 0 0 3px 3px;
            color: #666;
        }
    </style>

    <link rel=\\"preload\\" href=\\"/fonts/bootstrap-icons.woff2\\" as=\\"font\\" type=\\"font/woff2\\" crossorigin>

    <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"test-font.css\\" />
</head>
<body>

    <h1>test-font</h1>


        <div class=\\"preview\\">
            <span class=\\"inner\\">
                <b class=\\"tf tf-my-icon\\"></b>
            </span>
            <br>
            <span class='label'>my-icon</span>
        </div>


</body>
</html>
"
`;
